src/index.js

```diff
@@ -1,143 +1,270 @@
+        return '>=' + match[1] + ' <=' + match[2];
+    }
+    match = version.match(/(\d+\.\d+\.\d+)<v<=(\d+\.\d+\.\d+)/);
+    if (match) {
+        return '>' + match[1] + ' <=' + match[2];
+    }
+    match = version.match(/(\d+\.\d+\.\d+)<v<(\d+\.\d+\.\d+)/);
+    if (match) {
+        return '>' + match[1] + ' <' + match[2];
+    }
+    return version;
+};
 
 // Transform all Elm dependencies into the semver versions.
-var transformDependencies = function(deps){
-  var result = {}
-  Object.keys(deps).forEach(function(key) {
-    result[key] = getSemerVersion(deps[key].replace(/\s/g, ''))
-  })
-  return result
-}
+var transformDependencies = function (deps) {
+    var result = {};
+    Object.keys(deps).forEach(function (key) {
+        result[key] = getSemerVersion(deps[key].replace(/\s/g, ''));
+    });
+    return result;
+};
 
 // Get the dependencies for a given package and reference.
-var getDependencies = function(package, ref) {
-  return new Promise(function (fulfill, reject){
-    getPackageJson(package, ref)
-      .then(function(json){
-        fulfill(transformDependencies(json.dependencies))
-      })
-  })
-}
+var getDependencies = function (packageName, ref) {
+    return new Promise(function (fulfill, reject) {
+        getPackageJson(packageName, ref)
+            .then(function (json) {
+                fulfill(transformDependencies(json.dependencies));
+            })
+            .catch(function(err) {
+                reject(err);
+            });
+    });
+};
 
 
 // Get the contents of the elm-package.json of the given package and reference
-var getPackageJson = function(package, ref){
-  var packageUrl = 'https://github.com/' + package + '/raw/' + ref + '/elm-package.json'
+var getPackageJson = function (packageName, ref) {
+
+    // console.log("getPackageJson: package=" + package + ", ref=" + ref);
+
+    if (isSsh(packageName)) {
+        return getPackageJsonSsh(packageName, ref);
+    } else {
+        var packageUrl = 'https://github.com/' + packageName + '/raw/' + ref + '/elm-package.json';
+        return new Promise(function (fulfill, reject) {
+            request.get(packageUrl, function (error, response, body) {
+                if (error) {
+                    reject(error);
+                } else {
+                    fulfill(JSON.parse(body));
+                }
+            });
+        });
+    }
+};
+
+var getPackageJsonSsh = function(packageName, ref) {
+
+    // console.log("getPackageJsonSsh: urlAndPackage=" + JSON.stringify(urlAndPackage, null, "  ") + ", ref=" + ref);
+
+    return new Promise(function (fulfill, reject) {
+
+        var repoName = packageName.split('/')[1];
+
+        var host = repoHost(packageName);
+
+        var cmd = 'cd /tmp && rm -rf ' + repoName +
+            ' && git clone --no-checkout --depth 1 --branch ' +
+            ref + ' git@' + host +
+            ':' + packageName + '.git &&' +
+            'cd elm-cassie && git show HEAD:elm-package.json';
```

Comment:
I think you left a hard coded reference to the repo your testing the code with: `elm-cassie`